lib/rspec/core/example_group.rb

```diff
@@ -26,7 +26,7 @@ module Core
     # your examples defined in {MemoizedHelpers} and {Pending}.
     class ExampleGroup
       extend Hooks
-
+      extend FailFastHandler
```

Comment:
The way of implementing it you're suggesting sounds better, I will address it as soon as I can. Thank you for the great feedback and apologise for the noise, I am pretty unfamiliar with this codebase.